### PR TITLE
Add Connector for Azure OpenAI Text

### PR DESCRIPTION
**Pull request by JDPrabasha**
# Description

Fixes https://github.com/wso2-enterprise/choreo/issues/19293

Related Pull Requests
- https://github.com/ballerina-platform/openapi-connectors/pull/758
- https://github.com/ballerina-platform/openapi-connectors/pull/760

One line release note: 
- Adds connector for Azure OpenAI Text

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Ballerina Version: 2201.4.1
* Operating System: Ubuntu 22.04.1 LTS
* Java SDK: openjdk version "1.8.0_362"

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/azure.openai.text/Module.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+## Overview
+This is a generated connector from [Azure OpenAI Completions API](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#completions/) OpenAPI specification.
+
+The Azure  OpenAI Service REST API Completions Endpoint will generate one or more predicted completions based on a provided prompt. The service can also return the probabilities of alternative tokens at each position.
+
+## Prerequisites
+- Create an [Azure](https://azure.microsoft.com/en-us/features/azure-portal/) account
+- Create an [Azure OpenAI resource](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource)
+- Deploy an appropriate model within the resource by referring to [this guide](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource?pivots=web-portal#deploy-a-model)
```
Comment (anupama-pathirage) [inline openapi/azure.openai.text/Module.md:9]:
```suggestion
- Deploy an appropriate model within the resource by referring to [Deploy a model](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource?pivots=web-portal#deploy-a-model) guide.
```

File: openapi/azure.openai.text/Module.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        serviceUrl = serviceUrl
+    );
+    ```
+
+### Step 3: Invoke connector operation
+1. Now you can use the operations available within the connector. 
+
+>**Note:** that they are in the form of remote operations.
+
+    Following is an example on text completion using an OpenAI text-davinci-002 model:
```
Comment (anupama-pathirage) [inline openapi/azure.openai.text/Module.md:39]:
```suggestion
    Following is an example on text completion using an OpenAI `text-davinci-002` model:
```

File: openapi/azure.openai.text/Module.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+### Step 2: Create a new connector instance
+
+Create and initialize a `text:Client` with the obtained `apiKey` and a `serviceUrl` containing the deployed models.
+
+    ```ballerina
+    final text:Client textClient = check new (
+        config = {auth: {apiKey: apiKey}},
+        serviceUrl = serviceUrl
+    );
+    ```
```
Comment (anupama-pathirage) [inline openapi/azure.openai.text/Module.md:32]: in github preview mode of this file, code snippets are not properly highlighted,  I guess it is due the front tabs.

File: openapi/azure.openai.text/Module.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    text:Inline_response_200|error unionResult = check textClient->/deployments/["davinci"]/completions.post("2023-03-15-preview", completionsBody);
+
+    if (unionResult is text:Inline_response_200) {
+            io:println(unionResult);
+        } else {
+            io:println(unionResult);
+        }
+    }
+    ```
```
Comment (anupama-pathirage) [inline openapi/azure.openai.text/Module.md:61]: in github preview mode of this file, code snippets are not properly highlighted,  I guess it is due the front tabs.

File: openapi/azure.openai.text/Module.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    ```ballerina
+    public function main() returns error? {
+
+    final text:Client textClient = check new (
+        config = {auth: {apiKey: apiKey}},
+        serviceUrl = serviceUrl
+    );
+
+    text:Deploymentid_completions_body completionsBody = {
+        prompt: ""What is Ballerina?",
```
Comment (anupama-pathirage) [inline openapi/azure.openai.text/Module.md:50]:
```suggestion
        prompt: "What is Ballerina?",
```

Comment (JDPrabasha) [reply openapi/azure.openai.text/Module.md]: Addressed in 1e06b4a73a6fb85503f6e1d9f6994854226367f6

File: openapi/azure.openai.text/Module.md (hunk 3 above)
Comment (JDPrabasha) [reply openapi/azure.openai.text/Module.md]: Addressed in fc616ce40263376a40a09f576de689fd81540439

File: openapi/azure.openai.text/Module.md (hunk 4 above)
Comment (JDPrabasha) [reply openapi/azure.openai.text/Module.md:58]: Addressed in fc616ce40263376a40a09f576de689fd81540439

**Review by anupama-pathirage — APPROVED** [review]

**Pull request closed without merging**